### PR TITLE
Enable issue-url feature in color-eyre

### PR DESCRIPTION
**Pull request by yaahc**

### CODE REVIEW

File: zebrad/src/application.rs [hunk 1]
```diff
@@ -89,7 +89,11 @@ impl Application for ZebradApp {
         let terminal = Terminal::new(self.term_colors(command));
         // This MUST happen after `Terminal::new` to ensure our preferred panic
         // handler is the last one installed
-        color_eyre::install().unwrap();
+        color_eyre::config::HookBuilder::default()
+            .issue_url("https://github.com/ZcashFoundation/zebra/issues/new")
+            .add_issue_metadata("version", "3.0.0-alpha.0")
+            .install()
+            .unwrap();
```
Comment (hdevalence) [inline zebrad/src/application.rs]: woah

File: zebrad/src/application.rs [hunk 2]
```diff
@@ -89,7 +89,11 @@ impl Application for ZebradApp {
         let terminal = Terminal::new(self.term_colors(command));
         // This MUST happen after `Terminal::new` to ensure our preferred panic
         // handler is the last one installed
-        color_eyre::install().unwrap();
+        color_eyre::config::HookBuilder::default()
+            .issue_url("https://github.com/ZcashFoundation/zebra/issues/new")
+            .add_issue_metadata("version", "3.0.0-alpha.0")
```
Comment (yaahc) [inline zebrad/src/application.rs]: I haven't done it yet but I want to consider whether we should add metadata about the system as well to these issues. Here's the example this feature was based on where the dev also added the `os_type` and `os_release` to their issue template.

https://twitter.com/chrisbiscardi/status/1306341323348144128

**Comment by yaahc** [comment]
Tested by adding a panic to the signal handler to force an error when I <ctrl-c> zebra.

```
^CSep 17 13:29:50.084  INFO sig{kind=SignalKind(2) name="SIGINT"}: zebrad::signal: received SIGINT, starting shutdown
The application panicked (crashed).
Message:  testing error messages
Location: zebrad/src/components/tokio.rs:90

  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ SPANTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  
   0: zebrad::components::tokio::imp::sig with kind=SignalKind(2) name="SIGINT"
      at zebrad/src/components/tokio.rs:74

Backtrace omitted.
Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.

Consider reporting this error using this URL: https://github.com/ZcashFoundation/zebra/issues/new?title=%3Cautogenerated-issue%3E&body=%23%23+Error%0A%60%60%60%0Atesting+error+messages%0A%60%60%60%0A%0A%23%23+Metadata%0A%7Ckey%7Cvalue%7C%0A%7C--%7C--%7C%0A%7C**version**%7C3.0.0-alpha.0%7C%0A%7C**location**%7Czebrad%2Fsrc%2Fcomponents%2Ftokio.rs%3A90%3A9%7C%0A%0A%0A%23%23+SpanTrace%0A%0A%3Cdetails%3E%0A%0A%60%60%60%0ASpanTrace%3A%0A+++0%3A+zebrad%3A%3Acomponents%3A%3Atokio%3A%3Aimp%3A%3Asig%0A+++++++++++with+kind%3DSignalKind%282%29+name%3D%22SIGINT%22%0A+++++++++++++at+zebrad%2Fsrc%2Fcomponents%2Ftokio.rs%3A74%0A%60%60%60%0A%3C%2Fdetails%3E
Aborted (core dumped)
```

**Comment by yaahc** [comment]
clickable version of the link :D 

https://github.com/ZcashFoundation/zebra/issues/new?title=%3Cautogenerated-issue%3E&body=%23%23+Error%0A%60%60%60%0Atesting+error+messages%0A%60%60%60%0A%0A%23%23+Metadata%0A%7Ckey%7Cvalue%7C%0A%7C--%7C--%7C%0A%7C**version**%7C3.0.0-alpha.0%7C%0A%7C**location**%7Czebrad%2Fsrc%2Fcomponents%2Ftokio.rs%3A90%3A9%7C%0A%0A%0A%23%23+SpanTrace%0A%0A%3Cdetails%3E%0A%0A%60%60%60%0ASpanTrace%3A%0A+++0%3A+zebrad%3A%3Acomponents%3A%3Atokio%3A%3Aimp%3A%3Asig%0A+++++++++++with+kind%3DSignalKind%282%29+name%3D%22SIGINT%22%0A+++++++++++++at+zebrad%2Fsrc%2Fcomponents%2Ftokio.rs%3A74%0A%60%60%60%0A%3C%2Fdetails%3E

**Comment by yaahc** [comment]
its so gooood

Screenshot from 2020-09-17 13-35-04

**Review by hdevalence — DISMISSED** [review]
wow, this looks sick, let's try it out!!

in the future we might want to grab the commit hash during the build process and store it in a constant so that we can bundle it into the report metadata.

File: zebrad/src/application.rs (hunk 2 above)
Comment (hdevalence) [reply zebrad/src/application.rs]: :+1: seems like a good idea, but also something that we can add incrementally in future PRs?

Comment (yaahc) [reply zebrad/src/application.rs]: :100:

Comment (hawkw) [inline zebrad/src/application.rs]: drive-by note: you can get this automatically with 
```rust
env!("CARGO_PKG_VERSION")
```
instead of a hard-coded string

Comment (yaahc) [reply zebrad/src/application.rs]: blessed, ty, i tried and failed to find how to look this up yesterday.

Comment (hdevalence) [reply zebrad/src/application.rs]: nifty: https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates

Comment (hdevalence) [reply zebrad/src/application.rs]: `CARGO_PKG_REPOSITORY` might be useful too

**Review by hdevalence — DISMISSED** [review]

File: zebrad/src/application.rs (hunk 2 above)
Comment (hawkw) [reply zebrad/src/application.rs]: oh, yeah, you could do the `issue_url` with
```rust
concat!(env!("CARGO_PKG_REPOSITORY"), "/issues/new")
``` 
or something!

Comment (yaahc) [reply zebrad/src/application.rs]: this works and is extremely sick

**Review by hdevalence — APPROVED** [review]

**Comment by teor2345** [comment]
> in the future we might want to grab the commit hash during the build process and store it in a constant so that we can bundle it into the report metadata.

@hdevalence sounds like a great idea, here's a similar ticket where we talked about adding the hash to the version #692
